src/gui/title_bar.cpp

```diff
@@ -0,0 +1,600 @@
+	bool warned_double_dosbox    = false;
+	bool warned_double_program   = false;
+	bool warned_double_version   = false;
+
+	auto check_not_set_animation = [&]() {
+		if (config.animated_rec && !warned_double_animation) {
+			LOG_WARNING("SDL: 'animation' specified more than once in 'titlebar_content'");
+			warned_double_animation = true;
+		}
+	};
+
+	auto check_not_set_cycles = [&]() {
+		if (config.show_cycles && !warned_double_cycles) {
+			LOG_WARNING("SDL: 'cycles' specified more than once in 'titlebar_content'");
+			warned_double_cycles = true;
+		}
+	};
+
+	auto check_not_set_dosbox = [&]() {
+		if (config.show_dosbox && !warned_double_dosbox) {
+			LOG_WARNING("SDL: 'dosbox' specified more than once in 'titlebar_content'");
+			warned_double_dosbox = true;
+		}
+	};
+
+	auto check_not_set_program = [&]() {
+		if (config.program != ProgramDisplay::None && !warned_double_program) {
+			LOG_WARNING("SDL: 'program' specified more than once in 'titlebar_content'");
+			warned_double_program = true;
+		}
+	};
+
+	auto check_not_set_version = [&]() {
+		if (config.version != VersionDisplay::None && !warned_double_version) {
+			LOG_WARNING("SDL: 'version' specified more than once in 'titlebar_content'");
+			warned_double_version = true;
+		}
+	};
+
+	config = TitlebarConfig();
+
+	const auto titlebar_content = split(conf->Get_string("titlebar_content"));
+	for (const auto& setting : titlebar_content) {
+		if (iequals(setting, "animation")) {
+			check_not_set_animation();
+			config.animated_rec = true;
+			continue;
+		}
+
+		if (iequals(setting, "cycles")) {
+			check_not_set_cycles();
+			config.show_cycles = true;
+			continue;
+		}
+
+		if (iequals(setting, "dosbox")) {
+			check_not_set_dosbox();
+			config.show_dosbox = true;
+			continue;
+		}
+
+		if (iequals(setting, "program") ||
+		    iequals(setting, "program=name")) {
+			check_not_set_program();
+			config.program = ProgramDisplay::Name;
+			continue;
+		}
+
+		if (iequals(setting, "program=path")) {
+			check_not_set_program();
+			config.program = ProgramDisplay::Path;
+			continue;
+		}
+
+		if (iequals(setting, "program=segment")) {
+			check_not_set_program();
+			config.program = ProgramDisplay::Segment;
+			continue;
+		}
+
+		if (iequals(setting, "version") ||
+		    iequals(setting, "version=simple")) {
+			check_not_set_version();
+			config.version = VersionDisplay::Simple;
+			continue;
+		}
+
+		if (iequals(setting, "version=detailed")) {
+			check_not_set_version();
+			config.version = VersionDisplay::Detailed;
+			continue;
+		}
+
+		LOG_WARNING("SDL: Unknown setting '%s' in 'titlebar_content'",
+		            setting.c_str());
+	}
+
+	config.titlebar_text = conf->Get_string("titlebar_text");
+	trim(config.titlebar_text);
+}
+
+void TITLEBAR_AddConfig(Section_prop &secprop)
+{
+	constexpr auto always = Property::Changeable::Always;
+
+	Prop_string* prop_str = nullptr;
+
+	prop_str = secprop.Add_string("titlebar_content", always,
```

Comment:
Can we include include `window` in the setting's name? Can this just be `window_titlebar`?